cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -168,36 +167,18 @@ protected int getInstances(HttpRequest request) throws IllegalArgumentException,
     }
   }
 
-  protected final void getAppRecords(HttpResponder responder, Store store, String namespaceId, String appId) {
-    if (appId != null && appId.isEmpty()) {
-      responder.sendString(HttpResponseStatus.BAD_REQUEST, "app-id is empty");
-      return;
-    }
-
+  protected final void getAppRecords(HttpResponder responder, Store store, String namespaceId) {
     try {
-      Id.Namespace accId = Id.Namespace.from(namespaceId);
+      Id.Namespace namespace = Id.Namespace.from(namespaceId);
       List<ApplicationRecord> appRecords = Lists.newArrayList();
       List<ApplicationSpecification> specList;
-      if (appId == null) {
-        specList = new ArrayList<ApplicationSpecification>(store.getAllApplications(accId));
-      } else {
-        ApplicationSpecification appSpec = store.getApplication(new Id.Application(accId, appId));
-        if (appSpec == null) {
-          responder.sendStatus(HttpResponseStatus.NOT_FOUND);
-          return;
-        }
-        specList = Collections.singletonList(store.getApplication(new Id.Application(accId, appId)));
-      }
+      specList = new ArrayList<>(store.getAllApplications(namespace));
```

Comment:
In fact you should just be able to do the ``store.getAllApplications(namespace)`` in the for loop below